File: Convert Sorted Array to Binary Search Tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param num, a list of integers
    # @return a tree node
    def sortedArrayToBST(self, num):
        if (len(num) == 0):
            return None
        mid_pos = len(num) / 2
        left = num[0:mid_pos]
        mid = num[mid_pos]
        right = num[(mid_pos+1):len(num)]
        root = TreeNode(mid)
        root.left = self.sortedArrayToBST(left)
        root.right = self.sortedArrayToBST(right)
        return root

File: Convert Sorted Array to Binary Search Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode sortedArrayToBST(int[] num) {
        if (num.length == 0)
            return null;
        int mid_pos = num.length / 2;
        int[] left = java.util.Arrays.copyOfRange(num, 0, mid_pos);
        int mid = num[mid_pos];
        int[] right = java.util.Arrays.copyOfRange(num, mid_pos + 1, num.length);
        TreeNode root = new TreeNode(mid);
        root.left = sortedArrayToBST(left);
        root.right = sortedArrayToBST(right);
        return root;
    }
}
